### PR TITLE
fix: Add more tests for hive config loading

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix: Add more tests for hive config loading

We should not load args during our unit tests, because clap will try to parse the input for `cargo test`.

### CODE REVIEW

File: src/query/config/src/inner.rs [hunk 1]
```diff
@@ -71,6 +71,14 @@ impl Config {
         Ok(cfg)
     }
 
+    /// # NOTE
+    ///
+    /// This function is served for tests only.
+    pub fn load_without_args() -> Result<Self> {
```
Comment (ClSlaid) [inline src/query/config/src/inner.rs:77]: If this is only used in tests, can we add an `#[cfg(test)]` for it?

File: src/query/config/src/outer_v0.rs [hunk 2]
```diff
@@ -158,6 +158,40 @@ impl Config {
...
+
+    /// Load will load config from file, env and args.
+    ///
+    /// - Load from file as default.
+    /// - Load from env, will override config from file.
+    ///
+    /// # NOTE
+    ///
+    /// This function is only used for tests.
+    pub fn load_without_args() -> Result<Self> {
```
Comment (ClSlaid) [inline src/query/config/src/outer_v0.rs:170]: can this be configured with `#[cfg(test)]?

File: src/query/config/src/inner.rs (hunk 1 above)
Comment (Xuanwo) [reply src/query/config/src/inner.rs]:
```rust
error[E0599]: no function or associated item named `load_for_test` found for struct `databend_meta::configs::Config` in the current scope
  --> src/meta/service/tests/it/configs.rs:71:27
   |
71 |         let cfg = Config::load_for_test().expect("load must success");
   |                           ^^^^^^^^^^^^^ function or associated item not found in `databend_meta::configs::Config`
```

Not works for tests that not in the same crate.

Comment (ClSlaid) [reply src/query/config/src/inner.rs]: > Not works for tests that not in the same crate.

At least, we learn more about rust.

**Comment by BohuTANG** [comment]
We can do more unit tests for config, we can not do it because `clap will try to parse the input for cargo test` in the past.

**Pull request merged**